src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtSpoeAreasOfLawServiceTest.java

```diff
@@ -0,0 +1,166 @@
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static uk.gov.hmcts.dts.fact.util.TestHelper.getResourceAsJson;
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@ContextConfiguration(classes = AdminCourtSpoeAreasOfLawService.class)
+public class AdminCourtSpoeAreasOfLawServiceTest {
+
+    private static final String COURT_SLUG = "some slug";
+    private static final int COURT_AREAS_OF_LAW_COUNT = 3;
+    private static final String NOT_FOUND = "Not found: ";
+    private static final String DUPLICATE_ERROR = "Duplicate single point of entries exist";
+    private static final String TEST_COURT_AREAS_OF_LAW_PATH = "court-spoe-areas-of-law.json";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final List<CourtAreaOfLawSpoe> COURT_SPOE_AREA_OF_LAWS = new ArrayList<>();
+    private static final List<AreaOfLaw> DUPLICATED_SPOE_LIST = new ArrayList<>();
+
+    @MockBean
+    private CourtRepository courtRepository;
+
+
+    @MockBean
+    private AdminAuditService adminAuditService;
+
+    @MockBean
+    private CourtAreaOfLawSpoeRepository courtAreaOfLawSpoeRepository;
+
+    @Mock
+    private static Court court;
+
+    @Autowired
+    private AdminCourtSpoeAreasOfLawService adminCourtSpoeAreasOfLawService;
+
+    @BeforeAll
+    static void setUp() {
+        uk.gov.hmcts.dts.fact.entity.AreaOfLaw areaOfLawOne = new uk.gov.hmcts.dts.fact.entity.AreaOfLaw();
+        areaOfLawOne.setId(1);
+        areaOfLawOne.setName("AreaOfLaw1");
+        uk.gov.hmcts.dts.fact.entity.AreaOfLaw areaOfLawTwo = new uk.gov.hmcts.dts.fact.entity.AreaOfLaw();
+        areaOfLawTwo.setId(2);
+        areaOfLawTwo.setName("AreaOfLaw2");
+        uk.gov.hmcts.dts.fact.entity.AreaOfLaw areaOfLawThree = new uk.gov.hmcts.dts.fact.entity.AreaOfLaw();
+        areaOfLawThree.setId(3);
+        areaOfLawThree.setName("AreaOfLaw3");
+
+        COURT_SPOE_AREA_OF_LAWS.add(new CourtAreaOfLawSpoe(areaOfLawOne, court));
+        COURT_SPOE_AREA_OF_LAWS.add(new CourtAreaOfLawSpoe(areaOfLawTwo, court));
+        COURT_SPOE_AREA_OF_LAWS.add(new CourtAreaOfLawSpoe(areaOfLawThree, court));
+    }
+
+
+    @Test
+    void shouldReturnAllSpoeAreasOfLaw() {
+        when(courtAreaOfLawSpoeRepository.findAll()).thenReturn(COURT_SPOE_AREA_OF_LAWS);
+
+        assertThat(adminCourtSpoeAreasOfLawService.getAllSpoeAreasOfLaw())
+            .hasSize(COURT_AREAS_OF_LAW_COUNT)
+            .first()
+            .isInstanceOf(AreaOfLaw.class);
```

Comment:
should we check for any specific details within the area of law list element here?